operator/controllers/instance_controller.go

```diff
@@ -0,0 +1,461 @@
+		}
+		if err := r.Create(ctx, desiredServiceAccount); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+	}
+
+	currentRoleBinding := &rbacv1.RoleBinding{}
+	if err := r.Get(ctx, namespacedName, currentRoleBinding); err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+		desiredRoleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName, // The RoleBinding has the same name as the Pod.
+				Namespace: namespace,
+				// TODO: add proper labels.
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      "ServiceAccount",
+				Name:      podName,
+				Namespace: namespace,
+			}},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				APIGroup: "rbac.authorization.k8s.io",
+				Name:     "cluster-admin",
+			},
+		}
+		if err := ctrl.SetControllerReference(instance, desiredRoleBinding, r.Scheme); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+		if err := r.Create(ctx, desiredRoleBinding); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+	}
+
+	currentPod := &corev1.Pod{}
+	if err := r.Get(ctx, namespacedName, currentPod); err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+		desiredPod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: namespace,
+				// TODO: add proper labels.
+			},
+			Spec: corev1.PodSpec{
+				ServiceAccountName: podName,
+				RestartPolicy:      corev1.RestartPolicyNever,
+				Containers: []corev1.Container{{
+					Name:            "provisioning",
+					Image:           r.ProvisioningPodImage,
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					Command:         []string{"/bin/bash"},
+					Args:            []string{"-c", provisioningScript},
+					Env: []corev1.EnvVar{
+						{Name: "NAME", Value: helmInstanceName},
+						{Name: "CHART", Value: plan.Spec.Provisioning.Chart.URL},
+						{Name: "NAMESPACE", Value: instance.Namespace},
+					},
+					VolumeMounts: []corev1.VolumeMount{{
+						Name:      "values",
+						ReadOnly:  true,
+						MountPath: "/etc/metabroker-provisioning/",
+					}},
+				}},
+				Volumes: []corev1.Volume{{
+					Name: "values",
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{SecretName: valuesSecretName},
+					},
+				}},
+			},
+		}
+		if err := ctrl.SetControllerReference(instance, desiredPod, r.Scheme); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+		if err := r.Create(ctx, desiredPod); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to process provisioning pod: %w", err)
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if currentPod.Status.Phase == corev1.PodSucceeded {
```

Comment:
Please leave some kind of trace that this has happened; since the associated resources will go away, so will their logs.

Reply:
I think this belongs to the status of the `Instance`, which is still not implemented. Since there's a `TODO` in the status spec, I don't think there's an action item here.